File: backend/app/schemas/store_merchant.py
"""
商家入驻系统相关的数据模型
包含：商家认证、公告、活动、折扣、Banner
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum


# ==================== 枚举类型 ====================

class MerchantStatus(str, Enum):
    """商家认证状态"""
    PENDING = "PENDING"
    APPROVED = "APPROVED"
    REJECTED = "REJECTED"
    SUSPENDED = "SUSPENDED"


class MerchantLevel(str, Enum):
    """商家等级"""
    BASIC = "BASIC"
    PREMIUM = "PREMIUM"
    VIP = "VIP"


class ContentStatus(str, Enum):
    """内容状态"""
    DRAFT = "DRAFT"
    PUBLISHED = "PUBLISHED"
    HIDDEN = "HIDDEN"
    ENDED = "ENDED"


class ActivityType(str, Enum):
    """活动类型"""
    TRUNK_SHOW = "TRUNK_SHOW"
    POP_UP = "POP_UP"
    SALE = "SALE"
    EVENT = "EVENT"
    OTHER = "OTHER"


class DiscountType(str, Enum):
    """折扣类型"""
    PERCENTAGE = "PERCENTAGE"
    FIXED = "FIXED"
    SPECIAL = "SPECIAL"


class LinkType(str, Enum):
    """链接类型"""
    INTERNAL = "INTERNAL"
    EXTERNAL = "EXTERNAL"
    NONE = "NONE"


# ==================== 商家认证相关 ====================

class StoreMerchantCreate(BaseModel):
    """商家入驻申请"""
    storeId: str = Field(..., description="关联的买手店ID")
    contactName: Optional[str] = Field(None, max_length=100, description="联系人姓名")
    contactPhone: Optional[str] = Field(None, max_length=50, description="联系电话")
    contactEmail: Optional[str] = Field(None, max_length=200, description="联系邮箱")
    businessLicense: Optional[str] = Field(None, description="营业执照图片URL")


class StoreMerchantUpdate(BaseModel):
    """更新商家信息"""
    contactName: Optional[str] = Field(None, max_length=100)
    contactPhone: Optional[str] = Field(None, max_length=50)
    contactEmail: Optional[str] = Field(None, max_length=200)
    businessLicense: Optional[str] = None


class StoreMerchantReview(BaseModel):
    """审核商家申请"""
    status: MerchantStatus = Field(..., description="审核状态")
    rejectReason: Optional[str] = Field(None, description="拒绝原因")


class StoreMerchantAdminUpdate(BaseModel):
    """管理员更新商家信息"""
    status: Optional[MerchantStatus] = Field(None, description="商家状态")
    merchantLevel: Optional[MerchantLevel] = Field(None, description="商家等级")
    canPostBanner: Optional[bool] = Field(None, description="发布Banner权限")
    canPostAnnouncement: Optional[bool] = Field(None, description="发布公告权限")
    canPostActivity: Optional[bool] = Field(None, description="发布活动权限")
    canPostDiscount: Optional[bool] = Field(None, description="发布折扣权限")


class BuyerStoreUpdate(BaseModel):
    """商家更新店铺信息"""
    name: Optional[str] = Field(None, max_length=200, description="店铺名称")
    address: Optional[str] = Field(None, description="详细地址")
    phone: Optional[List[str]] = Field(None, description="联系电话列表")
    hours: Optional[str] = Field(None, max_length=100, description="营业时间")
    description: Optional[str] = Field(None, description="店铺描述")
    images: Optional[List[str]] = Field(None, description="店铺图片列表")
    rest: Optional[str] = Field(None, max_length=100, description="休息日信息")
    brands: Optional[List[str]] = Field(None, description="销售品牌列表")
    style: Optional[List[str]] = Field(None, description="风格标签列表")


class BuyerStore(BaseModel):
    """店铺信息"""
    id: str
    name: str
    address: str
    city: str
    country: str
    latitude: float
    longitude: float
    brands: List[str] = []
    style: List[str] = []
    isOpen: bool = True
    phone: List[str] = []
    hours: Optional[str] = None
    rating: Optional[float] = None
    description: Optional[str] = None
    images: List[str] = []
    rest: Optional[str] = None
    createdAt: Optional[str] = None
    updatedAt: Optional[str] = None


class StoreMerchant(BaseModel):
    """商家信息"""
    id: int
    storeId: str
    userId: int
    contactName: Optional[str] = None
    contactPhone: Optional[str] = None
    contactEmail: Optional[str] = None
    businessLicense: Optional[str] = None
    status: str
    rejectReason: Optional[str] = None
    reviewedBy: Optional[int] = None
    reviewedAt: Optional[str] = None
    merchantLevel: str = "BASIC"
    canPostBanner: bool = True
    canPostAnnouncement: bool = True
    canPostActivity: bool = True
    canPostDiscount: bool = True
    createdAt: str
    updatedAt: str


# ==================== 公告相关 ====================

class StoreAnnouncementCreate(BaseModel):
    """创建公告"""
    title: str = Field(..., min_length=1, max_length=200, description="公告标题")
    content: str = Field(..., min_length=1, description="公告内容")
    isPinned: bool = Field(default=False, description="是否置顶")
    status: ContentStatus = Field(default=ContentStatus.PUBLISHED, description="发布状态")
    startTime: Optional[str] = Field(None, description="开始时间")
    endTime: Optional[str] = Field(None, description="结束时间")


class StoreAnnouncementUpdate(BaseModel):
    """更新公告"""
    title: Optional[str] = Field(None, max_length=200)
    content: Optional[str] = None
    isPinned: Optional[bool] = None
    status: Optional[ContentStatus] = None
    startTime: Optional[str] = None
    endTime: Optional[str] = None


class StoreAnnouncement(BaseModel):
    """公告"""
    id: int
    storeId: str
    merchantId: int
    title: str
    content: str
    isPinned: bool = False
    status: str
    startTime: Optional[str] = None
    endTime: Optional[str] = None
    createdAt: str
    updatedAt: str


# ==================== Banner 相关 ====================

class StoreBannerCreate(BaseModel):
    """创建 Banner"""
    title: Optional[str] = Field(None, max_length=200, description="Banner标题")
    imageUrl: str = Field(..., description="图片URL")
    linkUrl: Optional[str] = Field(None, description="点击跳转链接")
    linkType: LinkType = Field(default=LinkType.NONE, description="链接类型")
    sortOrder: int = Field(default=0, description="排序")
    status: ContentStatus = Field(default=ContentStatus.PUBLISHED, description="发布状态")
    startTime: Optional[str] = Field(None, description="开始时间")
    endTime: Optional[str] = Field(None, description="结束时间")


class StoreBannerUpdate(BaseModel):
    """更新 Banner"""
    title: Optional[str] = Field(None, max_length=200)
    imageUrl: Optional[str] = None
    linkUrl: Optional[str] = None
    linkType: Optional[LinkType] = None
    sortOrder: Optional[int] = None
    status: Optional[ContentStatus] = None
    startTime: Optional[str] = None
    endTime: Optional[str] = None


class StoreBanner(BaseModel):
    """Banner"""
    id: int
    storeId: str
    merchantId: int
    title: Optional[str] = None
    imageUrl: str
    linkUrl: Optional[str] = None
    linkType: str = "NONE"
    sortOrder: int = 0
    status: str
    startTime: Optional[str] = None
    endTime: Optional[str] = None
    clickCount: int = 0
    viewCount: int = 0
    createdAt: str
    updatedAt: str


# ==================== 活动相关 ====================

class StoreActivityCreate(BaseModel):
    """创建活动"""
    title: str = Field(..., min_length=1, max_length=200, description="活动标题")
    description: Optional[str] = Field(None, description="活动描述")
    coverImage: Optional[str] = Field(None, description="封面图片")
    images: List[str] = Field(default=[], description="活动图片列表")
    activityStartTime: str = Field(..., description="活动开始时间")
    activityEndTime: str = Field(..., description="活动结束时间")
    location: Optional[str] = Field(None, max_length=500, description="活动地点")
    activityType: ActivityType = Field(default=ActivityType.EVENT, description="活动类型")
    status: ContentStatus = Field(default=ContentStatus.PUBLISHED, description="发布状态")
    needRegistration: bool = Field(default=False, description="是否需要报名")
    registrationLimit: Optional[int] = Field(None, description="报名人数限制")


class StoreActivityUpdate(BaseModel):
    """更新活动"""
    title: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    coverImage: Optional[str] = None
    images: Optional[List[str]] = None
    activityStartTime: Optional[str] = None
    activityEndTime: Optional[str] = None
    location: Optional[str] = Field(None, max_length=500)
    activityType: Optional[ActivityType] = None
    status: Optional[ContentStatus] = None
    needRegistration: Optional[bool] = None
    registrationLimit: Optional[int] = None


class StoreActivity(BaseModel):
    """活动"""
    id: int
    storeId: str
    merchantId: int
    title: str
    description: Optional[str] = None
    coverImage: Optional[str] = None
    images: List[str] = []
    activityStartTime: str
    activityEndTime: str
    location: Optional[str] = None
    activityType: str = "EVENT"
    status: str
    needRegistration: bool = False
    registrationLimit: Optional[int] = None
    registrationCount: int = 0
    createdAt: str
    updatedAt: str


# ==================== 折扣相关 ====================

class StoreDiscountCreate(BaseModel):
    """创建折扣"""
    title: str = Field(..., min_length=1, max_length=200, description="折扣标题")
    description: Optional[str] = Field(None, description="折扣描述")
    coverImage: Optional[str] = Field(None, description="封面图片")
    discountType: DiscountType = Field(..., description="折扣类型")
    discountValue: Optional[str] = Field(None, max_length=100, description="折扣值")
    applicableBrands: List[str] = Field(default=[], description="适用品牌")
    applicableCategories: List[str] = Field(default=[], description="适用品类")
    discountStartTime: str = Field(..., description="折扣开始时间")
    discountEndTime: str = Field(..., description="折扣结束时间")
    minPurchaseAmount: Optional[float] = Field(None, description="最低消费金额")
    termsAndConditions: Optional[str] = Field(None, description="使用条款")
    status: ContentStatus = Field(default=ContentStatus.PUBLISHED, description="发布状态")
    needCode: bool = Field(default=False, description="是否需要优惠码")
    discountCode: Optional[str] = Field(None, max_length=50, description="优惠码")


class StoreDiscountUpdate(BaseModel):
    """更新折扣"""
    title: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    coverImage: Optional[str] = None
    discountType: Optional[DiscountType] = None
    discountValue: Optional[str] = Field(None, max_length=100)
    applicableBrands: Optional[List[str]] = None
    applicableCategories: Optional[List[str]] = None
    discountStartTime: Optional[str] = None
    discountEndTime: Optional[str] = None
    minPurchaseAmount: Optional[float] = None
    termsAndConditions: Optional[str] = None
    status: Optional[ContentStatus] = None
    needCode: Optional[bool] = None
    discountCode: Optional[str] = Field(None, max_length=50)


class StoreDiscount(BaseModel):
    """折扣"""
    id: int
    storeId: str
    merchantId: int
    title: str
    description: Optional[str] = None
    coverImage: Optional[str] = None
    discountType: str
    discountValue: Optional[str] = None
    applicableBrands: List[str] = []
    applicableCategories: List[str] = []
    discountStartTime: str
    discountEndTime: str
    minPurchaseAmount: Optional[float] = None
    termsAndConditions: Optional[str] = None
    status: str
    needCode: bool = False
    discountCode: Optional[str] = None
    createdAt: str
    updatedAt: str


# ==================== 活动报名相关 ====================

class ActivityRegistrationCreate(BaseModel):
    """活动报名"""
    contactName: Optional[str] = Field(None, max_length=100, description="联系人姓名")
    contactPhone: Optional[str] = Field(None, max_length=50, description="联系电话")
    note: Optional[str] = Field(None, description="备注")


class ActivityRegistration(BaseModel):
    """活动报名信息"""
    id: int
    activityId: int
    userId: int
    username: Optional[str] = None
    userAvatar: Optional[str] = None
    contactName: Optional[str] = None
    contactPhone: Optional[str] = None
    note: Optional[str] = None
    status: str
    createdAt: str
    updatedAt: str


# ==================== 店铺商家信息汇总 ====================

class StoreMerchantContent(BaseModel):
    """店铺商家发布的所有内容汇总"""
    isMerchant: bool = False  # 该店铺是否有认证商家
    merchantInfo: Optional[StoreMerchant] = None
    banners: List[StoreBanner] = []
    announcements: List[StoreAnnouncement] = []
    activities: List[StoreActivity] = []
    discounts: List[StoreDiscount] = []

File: web/src/lib/services/store-merchant.ts
/**
 * Web 端 商家入驻 API 客户端.
 *
 * 1:1 对齐 frontend/src/services/storeMerchantService.ts —— 三端共用同一套
 * 后端契约 (`/api/store-merchants/*`):
 *   - 商家认证 (apply / my / by-store / update)
 *   - 公告 / Banner / 活动 / 折扣 CRUD
 *   - 商家侧店铺详情 (get / update buyer-store)
 *
 * 全部走 `apiClient`, 所以 401 刷新、envelope 解包、`Authorization` 注入
 * 都统一在一处. 这里只负责 URL / 参数 / 类型.
 */

import { apiClient } from "../api-client";

// ==================== 枚举类型 ====================

export type MerchantStatus = "PENDING" | "APPROVED" | "REJECTED" | "SUSPENDED";
export type MerchantLevel = "BASIC" | "PREMIUM" | "VIP";
export type ContentStatus = "DRAFT" | "PUBLISHED" | "HIDDEN" | "ENDED";
export type ActivityType =
  | "TRUNK_SHOW"
  | "POP_UP"
  | "SALE"
  | "EVENT"
  | "OTHER";
export type DiscountType = "PERCENTAGE" | "FIXED" | "SPECIAL";
export type LinkType = "INTERNAL" | "EXTERNAL" | "NONE";

// ==================== 商家 ====================

export interface StoreMerchant {
  id: number;
  storeId: string;
  userId: number;
  contactName?: string;
  contactPhone?: string;
  contactEmail?: string;
  businessLicense?: string;
  status: MerchantStatus;
  rejectReason?: string;
  reviewedBy?: number;
  reviewedAt?: string;
  merchantLevel: MerchantLevel;
  canPostBanner: boolean;
  canPostAnnouncement: boolean;
  canPostActivity: boolean;
  canPostDiscount: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface StoreMerchantUpdateParams {
  contactName?: string;
  contactPhone?: string;
  contactEmail?: string;
  businessLicense?: string;
}

// ==================== 公告 ====================

export interface StoreAnnouncement {
  id: number;
  storeId: string;
  merchantId: number;
  title: string;
  content: string;
  isPinned: boolean;
  status: ContentStatus;
  startTime?: string;
  endTime?: string;
  createdAt: string;
  updatedAt: string;
}

export interface StoreAnnouncementCreateParams {
  title: string;
  content: string;
  isPinned?: boolean;
  status?: ContentStatus;
  startTime?: string;
  endTime?: string;
}

export type StoreAnnouncementUpdateParams = Partial<
  StoreAnnouncementCreateParams
>;

// ==================== Banner ====================

export interface StoreBanner {
  id: number;
  storeId: string;
  merchantId: number;
  title?: string;
  imageUrl: string;
  linkUrl?: string;
  linkType: LinkType;
  sortOrder: number;
  status: ContentStatus;
  startTime?: string;
  endTime?: string;
  clickCount: number;
  viewCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface StoreBannerCreateParams {
  title?: string;
  imageUrl: string;
  linkUrl?: string;
  linkType?: LinkType;
  sortOrder?: number;
  status?: ContentStatus;
  startTime?: string;
  endTime?: string;
}

export type StoreBannerUpdateParams = Partial<StoreBannerCreateParams>;

// ==================== 活动 ====================

export interface StoreActivity {
  id: number;
  storeId: string;
  merchantId: number;
  title: string;
  description?: string;
  coverImage?: string;
  images: string[];
  activityStartTime: string;
  activityEndTime: string;
  location?: string;
  activityType: ActivityType;
  status: ContentStatus;
  needRegistration: boolean;
  registrationLimit?: number;
  registrationCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface StoreActivityCreateParams {
  title: string;
  description?: string;
  coverImage?: string;
  images?: string[];
  activityStartTime: string;
  activityEndTime: string;
  location?: string;
  activityType?: ActivityType;
  status?: ContentStatus;
  needRegistration?: boolean;
  registrationLimit?: number;
}

export type StoreActivityUpdateParams = Partial<StoreActivityCreateParams>;

// ==================== 折扣 ====================

export interface StoreDiscount {
  id: number;
  storeId: string;
  merchantId: number;
  title: string;
  description?: string;
  coverImage?: string;
  discountType: DiscountType;
  discountValue?: string;
  applicableBrands: string[];
  applicableCategories: string[];
  discountStartTime: string;
  discountEndTime: string;
  minPurchaseAmount?: number;
  termsAndConditions?: string;
  status: ContentStatus;
  needCode: boolean;
  discountCode?: string;
  createdAt: string;
  updatedAt: string;
}

export interface StoreDiscountCreateParams {
  title: string;
  description?: string;
  coverImage?: string;
  discountType: DiscountType;
  discountValue?: string;
  applicableBrands?: string[];
  applicableCategories?: string[];
  discountStartTime: string;
  discountEndTime: string;
  minPurchaseAmount?: number;
  termsAndConditions?: string;
  status?: ContentStatus;
  needCode?: boolean;
  discountCode?: string;
}

export type StoreDiscountUpdateParams = Partial<StoreDiscountCreateParams>;

// ==================== 店铺信息 (商家视角) ====================
//
// 注意: 后端 /api/store-merchants/buyer-store/{id} 返回的是扁平的 latitude /
// longitude (老接口), 而 `/api/buyer-stores/*` 返回的是嵌套 coordinates.
// 为了不破坏 Web 其它地方对 `BuyerStore` (coordinates 嵌套) 的约定, 单独给
// 这个商家端视图起个名字.

export interface MerchantBuyerStore {
  id: string;
  name: string;
  address: string;
  city: string;
  country: string;
  latitude: number;
  longitude: number;
  brands: string[];
  style: string[];
  isOpen: boolean;
  phone: string[];
  hours?: string;
  rating?: number;
  description?: string;
  images: string[];
  rest?: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface MerchantBuyerStoreUpdateParams {
  name?: string;
  address?: string;
  phone?: string[];
  hours?: string;
  description?: string;
  images?: string[];
  rest?: string;
  brands?: string[];
  style?: string[];
}

// ==================== 看板 / Insights ====================
//
// 商家后台「看板」页面响应类型。和 backend `store_insights_service` 1:1 对齐.

export interface InsightsOverview {
  wantToGoTotal: number;
  wantToGoToday: number;
  visitedTotal: number;
  visitedToday: number;
  ratingAverage: number;
  ratingCount: number;
}

export interface CityShare {
  city: string;
  count: number;
}

export interface HourBucket {
  hour: number;
  count: number;
}

export interface BrandShare {
  brandId: number;
  brandName: string;
  count: number;
}

export interface FanProfile {
  fansTotal: number;
  cityDistribution: CityShare[];
  activeHours: HourBucket[];
  preferredBrands: BrandShare[];
}

export interface TrendPoint {
  date: string;
  count: number;
}

export interface PromotionMetric {
  total: number;
  today: number;
  trend: TrendPoint[];
}

export interface PromotionStats {
  wantToGo: PromotionMetric;
  visited: PromotionMetric;
}

export interface VisitCommentReply {
  id: number;
  storeId: string;
  parentId: number;
  userId: number;
  username: string;
  userAvatar?: string;
  replyToUserId?: number;
  replyToUsername?: string;
  content: string;
  likeCount: number;
  createdAt: string;
  updatedAt: string;
}

export interface VisitComment {
  id: number;
  storeId: string;
  userId: number;
  username: string;
  userAvatar?: string;
  content: string;
  likeCount: number;
  replyCount: number;
  replies: VisitCommentReply[];
  createdAt: string;
  updatedAt: string;
}

interface VisitCommentListResponse {
  comments: VisitComment[];
  total: number;
  page: number;
  pageSize: number;
  storeId: string;
}

// 内容数据看板 V2 (品牌点击 & TOP 品牌)

/** 时间窗口允许 7 / 30 / 0(=全部) —— 与后端 ALLOWED_BRAND_WINDOWS 对齐. */
export type BrandStatsWindow = 7 | 30 | 0;

export interface BrandClickBreakdown {
  brand: string;
  wantCount: number;
  favoriteCount: number;
  likeCount: number;
  commentCount: number;
  viewCount: number;
  totalCount: number;
}

export interface BrandStatsResponse {
  windowDays: number;
  computedAt?: string | null;
  topBrands: BrandClickBreakdown[];
  allBrands: BrandClickBreakdown[];
}

export interface TopProductItem {
  id: number;
  title: string;
  brand?: string | null;
  coverImage?: string | null;
  wantCount: number;
  favoriteCount: number;
  likeCount: number;
  commentCount: number;
  viewCount: number;
  priceCents: number;
  currency: string;
  status: string;
}

interface TopProductsResponse {
  items: TopProductItem[];
  limit: number;
}

// ==================== 列表响应类型 ====================

interface MyMerchantsResponse {
  merchants: StoreMerchant[];
  total: number;
  page?: number;
  pageSize?: number;
}

interface AnnouncementListResponse {
  announcements: StoreAnnouncement[];
  total: number;
}

interface BannerListResponse {
  banners: StoreBanner[];
  total: number;
}

interface ActivityListResponse {
  activities: StoreActivity[];
  total: number;
}

interface DiscountListResponse {
  discounts: StoreDiscount[];
  total: number;
}

// ==================== API ====================

export const storeMerchantService = {
  // ── 商家认证 ─────────────────────────────────────────────────────────
  getMyMerchants: (page = 1, pageSize = 20) =>
    apiClient.get<MyMerchantsResponse>("/api/store-merchants/my", {
      page,
      pageSize,
    }),

  updateMerchant: (merchantId: number, data: StoreMerchantUpdateParams) =>
    apiClient.put<StoreMerchant>(`/api/store-merchants/${merchantId}`, data),

  // ── 店铺信息 (商家视角) ─────────────────────────────────────────────
  getBuyerStore: (storeId: string) =>
    apiClient.get<MerchantBuyerStore>(
      `/api/store-merchants/buyer-store/${encodeURIComponent(storeId)}`,
    ),

  updateBuyerStore: (storeId: string, data: MerchantBuyerStoreUpdateParams) =>
    apiClient.put<MerchantBuyerStore>(
      `/api/store-merchants/buyer-store/${encodeURIComponent(storeId)}`,
      data,
    ),

  // ── 公告 ─────────────────────────────────────────────────────────────
  getMerchantAnnouncements: (merchantId: number, page = 1, pageSize = 20) =>
    apiClient.get<AnnouncementListResponse>(
      `/api/store-merchants/${merchantId}/announcements`,
      { page, pageSize },
    ),

  createAnnouncement: (
    merchantId: number,
    data: StoreAnnouncementCreateParams,
  ) =>
    apiClient.post<StoreAnnouncement>(
      `/api/store-merchants/${merchantId}/announcements`,
      data,
    ),

  updateAnnouncement: (
    announcementId: number,
    data: StoreAnnouncementUpdateParams,
  ) =>
    apiClient.put<StoreAnnouncement>(
      `/api/store-merchants/announcements/${announcementId}`,
      data,
    ),

  deleteAnnouncement: (announcementId: number) =>
    apiClient.delete<void>(
      `/api/store-merchants/announcements/${announcementId}`,
    ),

  // ── Banner ──────────────────────────────────────────────────────────
  getMerchantBanners: (merchantId: number, page = 1, pageSize = 20) =>
    apiClient.get<BannerListResponse>(
      `/api/store-merchants/${merchantId}/banners`,
      { page, pageSize },
    ),

  createBanner: (merchantId: number, data: StoreBannerCreateParams) =>
    apiClient.post<StoreBanner>(
      `/api/store-merchants/${merchantId}/banners`,
      data,
    ),

  updateBanner: (bannerId: number, data: StoreBannerUpdateParams) =>
    apiClient.put<StoreBanner>(`/api/store-merchants/banners/${bannerId}`, data),

  deleteBanner: (bannerId: number) =>
    apiClient.delete<void>(`/api/store-merchants/banners/${bannerId}`),

  // ── 活动 ─────────────────────────────────────────────────────────────
  getMerchantActivities: (merchantId: number, page = 1, pageSize = 20) =>
    apiClient.get<ActivityListResponse>(
      `/api/store-merchants/${merchantId}/activities`,
      { page, pageSize },
    ),

  createActivity: (merchantId: number, data: StoreActivityCreateParams) =>
    apiClient.post<StoreActivity>(
      `/api/store-merchants/${merchantId}/activities`,
      data,
    ),

  updateActivity: (activityId: number, data: StoreActivityUpdateParams) =>
    apiClient.put<StoreActivity>(
      `/api/store-merchants/activities/${activityId}`,
      data,
    ),

  deleteActivity: (activityId: number) =>
    apiClient.delete<void>(`/api/store-merchants/activities/${activityId}`),

  // ── 折扣 ─────────────────────────────────────────────────────────────
  getMerchantDiscounts: (merchantId: number, page = 1, pageSize = 20) =>
    apiClient.get<DiscountListResponse>(
      `/api/store-merchants/${merchantId}/discounts`,
      { page, pageSize },
    ),

  createDiscount: (merchantId: number, data: StoreDiscountCreateParams) =>
    apiClient.post<StoreDiscount>(
      `/api/store-merchants/${merchantId}/discounts`,
      data,
    ),

  updateDiscount: (discountId: number, data: StoreDiscountUpdateParams) =>
    apiClient.put<StoreDiscount>(
      `/api/store-merchants/discounts/${discountId}`,
      data,
    ),

  deleteDiscount: (discountId: number) =>
    apiClient.delete<void>(`/api/store-merchants/discounts/${discountId}`),

  // ── 看板 / Insights ──────────────────────────────────────────────
  // 全部需要 merchant 本人鉴权，后端走 _ensure_merchant_owner。
  getInsightsOverview: (merchantId: number) =>
    apiClient.get<InsightsOverview>(
      `/api/store-merchants/${merchantId}/insights/overview`,
    ),

  getInsightsFans: (merchantId: number) =>
    apiClient.get<FanProfile>(
      `/api/store-merchants/${merchantId}/insights/fans`,
    ),

  getInsightsPromotion: (merchantId: number) =>
    apiClient.get<PromotionStats>(
      `/api/store-merchants/${merchantId}/insights/promotion`,
    ),

  getVisitComments: (merchantId: number, page = 1, pageSize = 20) =>
    apiClient.get<VisitCommentListResponse>(
      `/api/store-merchants/${merchantId}/insights/visit-comments`,
      { page, pageSize },
    ),

  getBrandStats: (
    merchantId: number,
    window: BrandStatsWindow = 7,
    topN = 3,
  ) =>
    apiClient.get<BrandStatsResponse>(
      `/api/store-merchants/${merchantId}/insights/brand-stats`,
      { window, topN },
    ),

  getTopProducts: (merchantId: number, limit = 10) =>
    apiClient.get<TopProductsResponse>(
      `/api/store-merchants/${merchantId}/insights/top-products`,
      { limit },
    ),
};

// ==================== 状态 / 枚举常量 ====================

export const MERCHANT_STATUS_LABEL: Record<MerchantStatus, string> = {
  PENDING: "审核中",
  APPROVED: "已认证",
  REJECTED: "已拒绝",
  SUSPENDED: "已暂停",
};

export const ACTIVITY_TYPE_LABEL: Record<ActivityType, string> = {
  TRUNK_SHOW: "Trunk Show",
  POP_UP: "快闪店",
  SALE: "特卖会",
  EVENT: "活动",
  OTHER: "其他",
};

export const DISCOUNT_TYPE_LABEL: Record<DiscountType, string> = {
  PERCENTAGE: "折扣比例",
  FIXED: "满减优惠",
  SPECIAL: "特别优惠",
};

export const CONTENT_STATUS_LABEL: Record<ContentStatus, string> = {
  DRAFT: "草稿",
  PUBLISHED: "已发布",
  HIDDEN: "已隐藏",
  ENDED: "已结束",
};
